test/unit/rules/require-button-type-test.js

```diff
@@ -142,18 +140,19 @@ generateRuleTests({
       fixedTemplate: '<form><button type="submit"></button></form>',
     },
     {
-      template: '/** silly example <button> usage */ <button><button>',
+      template: '<template>/** silly example <button/> usage */ <button></button></template>',
       meta: {
         filePath: 'layout.gjs',
       },
-      fixedTemplate: '/** silly example <button> usage */ <button type="button"></button>',
+      fixedTemplate:
+        '<template>/** silly example <button type="button" /> usage */ <button type="button"></button></template>',
```

Comment:
This is proving my point in the other comment. We can't use `currentSource.replace`. The `button` in the comment should never be touched.